source/common/stats/thread_local_store.cc

```diff
@@ -457,7 +457,24 @@ Gauge& ThreadLocalStoreImpl::ScopeImpl::gaugeFromStatName(StatName name,
   return gauge;
 }
 
+SymbolTable::StoragePtr
```

Comment:
@jmarantz the way I'm constructing the full stat name here seems to work at the surface, but when I try to fetch the stats back from the store it fails to find the existing stat, and it looks like the stat names in the store are garbage (a symbolTable().toString ends up printing out of bounds). Is there anything obviously wrong that I'm doing here? 

I kinda suspect that the storage isn't living long enough to be valid after I exit this scope, but not sure how else to create the symbol names.